presentation/common/src/presentation-common/PresentationRpcInterface.ts

```diff
@@ -186,7 +186,15 @@ export class PresentationRpcInterface extends RpcInterface {
   public async getContentSources(_token: IModelRpcProps, _options: ContentSourcesRpcRequestOptions): PresentationRpcResponse<ContentSourcesRpcResult> { return this.forward(arguments); }
 
   @RpcOperation.setPolicy({ allowResponseCompression: true })
-  public async getContentDescriptor(_token: IModelRpcProps, _options: ContentDescriptorRpcRequestOptions): PresentationRpcResponse<DescriptorJSON | undefined> { return this.forward(arguments); }
+  public async getContentDescriptor(_token: IModelRpcProps, _options: ContentDescriptorRpcRequestOptions): PresentationRpcResponse<DescriptorJSON | undefined> {
+    arguments[1] = { ...arguments[1], transport: "unparsed-json" };
+    const response: PresentationRpcResponseData<DescriptorJSON | string | undefined> = await this.forward(arguments);
+    if (response.statusCode === PresentationStatus.Success && typeof response.result === "string") {
+      response.result = JSON.parse(response.result);
+    }
+
+    return response as PresentationRpcResponseData<DescriptorJSON | undefined>;
+  }
```

Comment:
The initial plan was to bypass deserialisation and serialisation on other presentation RPC operations too, but I don't think that that's the right call. The current implementation of this optimisation, while shaves off a few seconds on large `getContentDescriptor` responses, introduces type casts that misrepresent the actual JavaScript data.

A proper implementation would involve an enhancement to RPC framework that adds support for streaming responses from native addon directly to frontend.